rasa/core/agent.py

```diff
@@ -336,7 +336,7 @@ def update_model(
         interpreter: Optional[NaturalLanguageInterpreter] = None,
         model_directory: Optional[Text] = None,
     ) -> None:
-        self.domain = domain
+        self.domain = self._create_domain(domain)
```

Comment:
Can you update the type annotation of `_create_domain`? It currently only accepts `Domainn` or `Text`. But `domain` can also be `None`.